### ISSUE TITLE
More monophyly flexibility

### ISSUE THREAD

**Issue by Anaphory** [lmaurits/BEASTling#151]
Given that the definition of monophyletic groups relies on named clades anyway, it would make sense to specify the required monophyly restrictions not only through monophyly levers, but optionally also through explicitly giving clade names.

It should probably impose the constraint on groups specified using eg.

    [languages]
    clades = west2793, insu1254

in addition to the (possibly empty) set of restrictions imposed using levels.

**Comment by lmaurits** [comment]
I like this idea!  Perhaps with the syntax `monophyly_clades = x, y, z`, for the sake of consistency.

**Comment by lmaurits** [comment]
Did you have strong desires/expectations on how much constraint this should pull in?  You used west2793, West Germanic, as an example, should this just require that all West Germanic languages be in one unstructured clade (i.e. (eng, deu, nld, afr)), or should this be recursive and also pull in constraints on Franconian, High German, North Sea Germanic (i.e. (eng, (deu,(nld,afr)))?

**Comment by Anaphory** [comment]
What I meant was “west2793 should be monophyletic”. The lower constraints could then be specified explicitly, but that does not work the other way round.

**Comment by lmaurits** [comment]
Good point that the minimal interpretation is more flexible (i.e. you can build up the maximal interpretation by listing multiple clades), but I would hate to have to, say, constrain all of Indo-European this way.

I am trying to decide how I feel about some kind of extra syntax for this, e.g. what if `west2793` resulted in (eng, deu, nld, afr) and `west2793+` resulted in (eng, (deu,(nld,afr))?

**Comment by Anaphory** [comment]
Hm, do you want to look at newick-based monophyly constraints first (that was the idea, wasn't it?) and then see what kind of thing might naturally make most sense starting from there? I mean, it might turn out that there is some super simple pattern matching that makes sense with other things for `((west2793))` vs. `west2793` or whatever.

**Comment by lmaurits** [comment]
Sorry, are you talking about waiting until after #162 so that constraints are derived internally from Glottolog newick trees?  Or are you talking about the user being able to provide their own constraints in newick format?

**Comment by Anaphory** [comment]
Both, I guess? #162 should teach us how to nicely use Newick trees internally to specify constraints. From there it's a small step to provide the user with an option to specify their own trees, and if that way of specifying gains a nice and general Glottolog fallback, that might inform a natural way of specifying `x and its subtree` or just `x`. Am I just adding unnecessary complications?

**Comment by lmaurits** [comment]
It's already possible to specify your own monophyly tree, with the `monophyly_newick` option in `[languages]`, I thought the point of this Issue was to provide some nicer alternative to this for simple cases, so the user doesn't have to wrangle newick just to choose a clean subset of Glottolog?

**Comment by Anaphory** [comment]
It may have been, heck, I'm really not up to speed with Beastling at the moment, spending far too more thought on CLDF.

In that case, you are essentially right, but the tiny `+` is too inconspicuous for that kind of magic in my opinion, any other suggestions?

**Comment by lmaurits** [comment]
No worries, I was modelling you as being aware of this because you were the one who opened the Issue I closed recently pointing out that `monophyly_newick` was undocumented, but I was perhaps failing to take into account that you probably opened it about a year ago!

I agree that `+` is too easy to overlook.  Something like `foo(west2793)` is far harder to miss, not sure what `foo` should be, though...

**Comment by Anaphory** [comment]
`subclades(west2736)`, or subtree or something like that?
